code/modules/lighting/lighting_verbs.dm

```diff
@@ -0,0 +1,102 @@
+#define CHECK_RIGHTS(PERM)     \
+	if (!check_rights(PERM)) { \
+		src << span("alert", "You do not have sufficient rights to perform this action.");\
+		return;                \
+	}
+
+var/list/admin_verbs_lighting = list(
+	/client/proc/lighting_hide_verbs,
+	/client/proc/lighting_flush,
+	/client/proc/lighting_reconsider_target,
+	/client/proc/lighting_build_overlay,
+	/client/proc/lighting_clear_overlay,
+	/client/proc/lighting_toggle_profiling
+)
+
+/client/proc/lighting_show_verbs()
+	set category = "Debug"
+	set name = "Show Lighting Verbs"
+	set desc = "Shows the lighting debug verbs."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+
+	src << span("notice", "Lighting debug verbs have been shown.")
+	verbs += admin_verbs_lighting
+
+/client/proc/lighting_hide_verbs()
+	set category = "Lighting"
+	set name = "Hide Lighting Verbs"
+	set desc = "Hides the lighting debug verbs."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+
+	src << span("notice", "Lighting debug verbs have been hidden.")
+	verbs -= admin_verbs_lighting
+
+/client/proc/lighting_flush()
+	set category = "Lighting"
+	set name = "Flush Work Queue"
+	set desc = "Flushes the lighting processor's current work queue."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+
+	if (alert("Flush Lighting Work Queue? This will invalidate all pending lighting updates.", "Reset Lighting", "No", "No", "Yes") != "Yes")
+		return
+
+	log_and_message_admins("has flushed the lighting processor queues.")
+	lighting_update_lights = list()
+	lighting_update_corners = list()
+	lighting_update_overlays = list()
+
+/client/proc/lighting_reconsider_target(turf/T in world)
+	set category = "Lighting"
+	set name = "Reconsider Visibility"
+	set desc = "Triggers a visibility update for a turf."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+
+	if (!T.dynamic_lighting)
+		src << "That turf is not dynamically lit."
+		return
+
+	T.reconsider_lights()
+
+/client/proc/lighting_build_overlay(turf/T in world)
+	set category = "Lighting"
+	set name = "Build Overlay"
+	set desc = "Builds a lighting overlay for a turf if it does not have one."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+
+	if (T.lighting_overlay)
+		src << "That turf already has a lighting overlay."
+		return
+
+	T.lighting_build_overlay()
+
+/client/proc/lighting_clear_overlay(turf/T in world)
+	set category = "Lighting"
+	set name = "Clear Overlay"
+	set desc = "Clears a lighting overlay for a turf if it has one."
+
+	CHECK_RIGHTS(R_DEBUG|R_DEV)
+		
+	if (!T.lighting_overlay)
+		src << "That turf doesn't have a lighting overlay."
```

Comment:
Needs a server log. Devs removing lighting overlays to see stuff they shouldn't be able to when.